internal/vm/importers/glob.go

```diff
@@ -0,0 +1,289 @@
+package importers
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"net/url"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/google/go-jsonnet"
+	"github.com/pkg/errors"
+)
+
+const (
+	globParamDirs           = "dirs"
+	globParamVerb           = "verb"
+	globParamStripExtension = "strip-extension"
+)
+
+var supportedParams = map[string]bool{
+	globParamDirs:           true,
+	globParamVerb:           true,
+	globParamStripExtension: true,
+}
+
+var supportedParamsDisplay string
+
+func init() {
+	var ret []string
+	for k := range supportedParams {
+		ret = append(ret, k)
+	}
+	sort.Strings(ret)
+	supportedParamsDisplay = strings.Join(ret, ", ")
+}
+
+// globImportParams are additional parameters for the glob import expressed as URI query params.
+type globImportParams struct {
+	verb           string // one of 'import' or 'importstr' - the verb to use to import files found
+	dirs           int    // number of directories levels to retain in the returned object (default: all)
+	stripExtension bool   // whether extensions should be stripped from the base name of the file in the map key
+}
+
+// keyFor returns the key to be used for the supplied file based on whether directory levels should be limited
+// and/ or extensions stripped.
+func (p globImportParams) keyFor(file string) string {
+	// short-circuit simple case
+	if p.dirs < 0 && !p.stripExtension {
+		return file
+	}
+	elements := strings.Split(file, string(filepath.Separator))
+	name := elements[len(elements)-1]
+	dirs := elements[0 : len(elements)-1]
+	if p.dirs >= 0 {
+		if len(dirs) > p.dirs {
+			dirs = dirs[len(dirs)-p.dirs:]
+		}
+	}
```

Comment:
This might result in aliasing names, right? It will produce a quite ugly runtime error then (about duplicate fields in an object). Not sure if that's by design.

Reply:
yes, the alias case is checked explicitly and causes a runtime error by design.